sentry-core/src/main/java/io/sentry/core/exception/ExceptionMechanism.java

```diff
@@ -0,0 +1,42 @@
+package io.sentry.core.exception;
+
+import io.sentry.core.protocol.Mechanism;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A throwable decorator that holds an {@link io.sentry.core.protocol.Mechanism} related to the
+ * decorated {@link Throwable}.
+ */
+public final class ExceptionMechanism extends RuntimeException {
+  private static final long serialVersionUID = 142345454265713915L;
+
+  private final Mechanism exceptionMechanism;
+  private final Throwable throwable;
+  private final Thread thread;
+
+  /**
+   * A {@link Throwable} that decorates another with a Sentry {@link Mechanism}.
+   *
+   * @param mechanism The {@link Mechanism}.
+   * @param throwable The {@link java.lang.Throwable}.
+   * @param thread The {@link java.lang.Thread}.
+   */
+  public ExceptionMechanism(
+      @Nullable Mechanism mechanism, @Nullable Throwable throwable, @Nullable Thread thread) {
```

Comment:
Before this was a throwable wrapping a throwable.
Now this is an exception that wraps a throwable.
Would we want this to wrap an exception instead?
```suggestion
      @Nullable Mechanism mechanism, @Nullable Exception exception, @Nullable Thread thread) {
```

Also, considering this is a decorator, we could name it `MechanismDecoratorException` or something along these lines?

Reply:
This file was added? What happened with the original `ExceptionMechanismThrowable`? Could it be a rename operation (delete the original file)?

Reply:
guess the IDE did something wrong, I just renamed it but fixed it.